20180224_kubernetes/index.html

```diff
@@ -0,0 +1,223 @@
+        <p>Pods run on <b>Nodes</b> <span class="thoughts">(worker machine)</span></p>
+        <img src="static/nodes.svg" />
+        <div class="notes">
+            A node is a worker machine in Kubernetes and may be a VM or physical machine, depending on the cluster.<br />
+            You might have heard of master/minions before, and nodes were previously called minions.
+            Multiple Pods can run on one Node.<br />
+            Every Kubernetes Node runs at least:
+            <ul>
+                <li>Kubelet, a process responsible for communication between the Kubernetes Master and the Node; it manages the Pods and the containers running on a machine.</li>
+                <li>A container runtime (like Docker, rkt) responsible for pulling the container image from a registry, unpacking the container, and running the application.</li>
+                <li>Containers should only be scheduled together in a single Pod if they are tightly coupled and need to share resources such as disk.</li>
+            </ul>
+        </div>
+    </div>
+
+    <div id="kubelet" class="step" data-x="3600" data-y="1600" data-z="-15000" data-rotate="300" data-scale="0.2">
+        <h1>Kubelet</h1>
+        <q>An agent that runs on each node in the cluster. It makes sure that containers* are running and healthy in a pod</q><br />
+        <span class="footnote">* as described in Pod specification files</span>
+    </div>
+
+    <div id="master" class="step" data-x="4200" data-y="-2500" data-z="0" data-rotate="270" data-scale="5">
+        <p>These <b class="positioning">Minion</b> nodes are controlled by a <b class="rotating">Master</b> node, running master components*:</p>
+        <div class="footnotes">
+            <b class="scaling">kube-apiserver</b>|<b class="scaling">etcd</b>|<b class="scaling">kube-scheduler</b>|<b class="scaling">kube-controller-manager</b>|<b class="scaling">cloud-controller-manager</b>
+            <p>* It's ok to ignore these components, GCP even abstracted them</p>
+        </div>
+    </div>
+
+    <div id="cluster" class="step" data-x="7500" data-y="-1000" data-scale="6">
+        <p>This set of <b>master</b> components and <b>worker nodes</b> make up the<br /> <b class="cluster">cluster</b></p>
+        <img src="static/cluster.svg" />
+    </div>
+
+    <div id="selectors" class="step" data-x="8300" data-y="2500" data-rotate="20" data-scale="4">
+        <p>Wait, there's more!</p>
+        <q>You can group pods, no matter on which node they are running, through <b>labels</b> and <b>selectors</b>!</q>
+    </div>
+
+    <div id="services" class="step" data-x="6000" data-y="4000" data-scale="2">
+        <p>Enter <b>Services</b> and <b>Controllers</b></p>
+    </div>
+
+    <div id="summary" class="step" data-x="6200" data-y="5400" data-z="-100" data-rotate-x="-40" data-rotate-y="10" data-scale="2">
+        <q>Let's summarize what we've learnt</q>
+        <img src="static/summary.png" />
```

Comment:
The summary png doesn't highlight the vocabulary introduced (or seems to use other vocabulary).